afs/afs-core/src/test/java/com/powsybl/afs/DependencyCacheTest.java

```diff
@@ -49,6 +49,8 @@ public TicBuilder setName(String name) {
         @Override
         public Tic build() {
             NodeInfo info = context.getStorage().createNode(context.getFolderInfo().getId(), name, "TIC", "", 0, new NodeGenericMetadata());
+            info.consistent();
+            context.getStorage().consistent(info.getId());
```

Comment:
It's not obvious if those 2 lines duplicate actions or not.
But I think that if we don't call info.consistent(), indeed the NodeInfo passed to Tic constructor is not up to date with node info from the app storage.

It raises the questions: should we call those 2 methods in all builders ? Maybe there is something to think in more depth here.

Reply:
we can refactor the method by passing the nodeInfo object to setConsitent method but i think this is not a good alternative because the method signature will be  a little bit  weird compare to other API's methods .
Proposition : we can change createNode mehtod signature to accept an additional parameter : consistent parameter. User in that case set explicitly consistent node's attribute  in creation time, so he's aware for what the consistency is. this solution may solve the problem of forgetting the setConsistent method call at the end of builder process.
@geofjamg  @sylvlecl  what do you think ?

Reply:
I think changing the createNode is not the solution, because we need to change the consistency **after** the creation.
A possibility would be to have the `setConsistent(String)` method return an up-to-date `NodeInfo` and use it, but it's not that good (you can still use the previous NodeInfo).

I think anyway the `isConsistent` flag should only be used by the AppStorage implementation, not by business layer objects (ProjectFile etc.).
If we really wanted to prevent its use elsewhere, maybe we should have a more restricted object inside class `AbstractNodeBase`. Or we juste say in javadoc that it should not be used outside `AppStorage` implementations :)

Reply:
Maybe we should make the consistent getter and setter package private ?
Then AppStorage implementation could use public or protected methods or predicates which allows them to perform the filtering.

It would protect the flag to be used elsewhere.

Reply:
> I think changing the createNode is not the solution, because we need to change the consistency **after** the creation.
i don't agree because sometimes we have to get a consistent node just after creation (that case) and sometimes not (in builders). so the new parameter consistent in create node method can be a solution to set the state of a node in creation time.
so here for example we obtain : 
            NodeInfo info = context.getStorage().createNode(context.getFolderInfo().getId(), name, "TIC", "", 0, new NodeGenericMetadata(), true);
instead of : 
            NodeInfo info = context.getStorage().createNode(context.getFolderInfo().getId(), name, "TIC", "", 0, new NodeGenericMetadata());
            info.consistent();
            context.getStorage().consistent(info.getId());

Reply:
> It's not obvious if those 2 lines duplicate actions or not.
> But I think that if we don't call info.consistent(), indeed the NodeInfo passed to Tic constructor is not up to date with node info from the app storage.
> 
> It raises the questions: should we call those 2 methods in all builders ? Maybe there is something to think in more depth here.

i don't agree because sometimes we have to get a consistent node just after creation (that case) and sometimes not (in builders). so the new parameter consistent in create node method can be a solution to set the state of a node in creation time.
so here for example we obtain :
NodeInfo info = context.getStorage().createNode(context.getFolderInfo().getId(), name, "TIC", "", 0, new NodeGenericMetadata(), true);
instead of :
NodeInfo info = context.getStorage().createNode(context.getFolderInfo().getId(), name, "TIC", "", 0, new NodeGenericMetadata());
info.consistent();
context.getStorage().consistent(info.getId());

Reply:
Yes you're right, it can be useful.
However it does not solve other cases where we need to dissociate creation and "finalization" :(